### ISSUE TITLE
Feature Request: Add description to `matchesPredicate` assertion

### ISSUE THREAD

**Issue by jdsee** [willowtreeapps/assertk#514]
Hi 👋 

It would be great if the `matchesPredicate` assertion would allow a description parameter to be displayed on an assertion error instead of _satisfy the predicate_.

It could look something like this:
```kotlin
assertThat(x).matchesPredicate("some very special requirements") { it.isSpecial() }
```
which would then result in the error message:
```
expected X(special=false) to satisfy some very special requirements
```

**Comment by evant** [comment]
this already exists no?
https://github.com/willowtreeapps/assertk/blob/main/assertk/src/commonMain/kotlin/assertk/assertions/predicate.kt

**Comment by jdsee** [comment]
Yes, the predicate matcher already exists. My request is to accept an optional description to improve the currently hardcoded error message.

Or am I still missing something? 😃

**Comment by evant** [comment]
*Ah* gotcha, I'm a little hesitant to add assertion-specific message overriding. There is some limited ability to override the message, for example you could set the name so that:

```kotlin
assertThat(12, name = "divisible by 5").matchesPredicate { it % 5 }
```

would give you

```
org.opentest4j.AssertionFailedError: divisible by 5
    expected 12 to satisfy the predicate
```

otherwise, I would really recommend using a custom assertion, where you can set the message exactly how you want

```kotlin
fun Assert<X>.isSpecial() = given { actual ->
    if (!it.isSpecial()) expected("${actual} to satisfy some very special requirements")
}

assertThat(x).isSpecial()
```

**Comment by saket** [comment]
I'm migrating my project to assertk and I have the same feature request. How do I replace my `Truth.assertWithMessage()` calls? I was using them as a way to describe my assertions. Here's an example:

https://github.com/saket/telephoto/blob/bdff7e6b09fb3a9785a3b54e0455e6fab6108539/zoomable-image/core/src/androidTest/kotlin/me/saket/telephoto/zoomable/ZoomableImageTest.kt#L610-L612

Using custom assertions to include messages feels a bit odd.

**Comment by saket** [comment]
Maybe I can use `Assert.all(message)` as a stop-gap?

**Comment by evant** [comment]
yeah if something is going to be added it would be more general like `assertWithMessage()`, there's been a bit of discussion on added to/customizing the error message but nothing's been settled on.

**Comment by saket** [comment]
Subscribed to #352!
